### ISSUE TITLE
Firefox do not start on Android after a while

### ISSUE THREAD

**Issue by soulgalore** [mozilla/geckodriver#1817]
## System

* Version: 0.28.0 (and 0.27.0)
* Platform: macOS 10.15.5 
* Firefox: 83 (on Android)
* Selenium: 4.0.0-alpha.7

## Background
At Wikimedia we run a performance device lab with Android phones. When we use Firefox we got problem starting Firefox. I can see the problem on all phones (we have five phones) and tried with different Firefox versions and the same thing. However I cannot reproduce it on my own machine at home, so it is probably something going on in the environment where we have the phones. We continuously run tests throughout the day and after a while Firefox cannot start. After a while it works again.I've talked with @whimboo about this on the Browsertime Slack channel.

I've worked around this by adding more retries and longer time between the tries. At the moment we try to start Firefox 10 times with a sleep time of 180 seconds between tries. When the Firefox can't start, it is like that for 6-7-8 tries and then starts to work again.

From the Geckodriver side the problem looks like this (through browsertime):
```
[2020-12-08 02:08:43] INFO: firefox failed to start, trying 9 more time(s): Socket timeout reading Marionette handshake data: Resource temporarily unavailable (os error 35)
[2020-12-08 02:08:43] INFO: Skip install HAR trigger on Android
[2020-12-08 02:10:02] INFO: firefox failed to start, trying 8 more time(s): Connection refused (os error 61)
[2020-12-08 02:10:02] INFO: Skip install HAR trigger on Android
[2020-12-08 02:11:20] INFO: firefox failed to start, trying 7 more time(s): Socket timeout reading Marionette handshake data: Resource temporarily unavailable (os error 35)
```

From Marionette it looks like this (and continue to try new ports until the timeout of 60s to start the browser):
```
10-19 13:30:49.684 11024 11053 I Gecko   : 1603110649684	Marionette	INFO	Listening on port 2829
10-19 13:30:49.684 11024 11053 I Gecko   : 1603110649684	Marionette	DEBUG	Marionette is listening
10-19 13:30:49.870 11024 11053 I Gecko   : 1603110649870	Marionette	DEBUG	Accepted connection 0 from 127.0.0.1:42033
10-19 13:30:49.927 11024 11053 I Gecko   : 1603110649927	Marionette	DEBUG	Closed connection 0
10-19 13:30:50.129 11024 11053 I Gecko   : 1603110650129	Marionette	DEBUG	Accepted connection 1 from 127.0.0.1:46354
10-19 13:30:50.228 11024 11053 I Gecko   : 1603110650228	Marionette	DEBUG	Closed connection 1
```
See the full log: https://gist.github.com/soulgalore/fa66184c767d07da10016148f8cb8345

I understand  that this is probably a problem in the environment and I wonder if there could be some more helpful logs from Geckodriver/Marionette to easier find the root problem? 

One thing that would help is to add a feature to Geckodriver to log the log to a file (`--logFile`). At the moment using Selenium the Geckodriver log is pushed to stdout at its really hard to keep track of and since these errors only happens sometimes it would be very useful to just log Geckodriver log to a file, so that we could always have the log turned on and just remove the file when thing works.

**Comment by whimboo** [comment]
@soulgalore can you please check if that is still a problem with the geckodriver 0.29.1 release? Thanks.

**Comment by soulgalore** [comment]
Yes, I'll do that, I'm gonna upgrade the 19th of April when I'm back and let you know.

**Comment by soulgalore** [comment]
So we are changing provider of running our tests (we will host in in-house) that's why I never got back.

**Comment by whimboo** [comment]
@soulgalore did you ever face this issue again? If not we should close as dupe of issue #1903. Thanks.
